WooCommerce/Classes/POS/Presentation/Reusable Views/Buttons/POSButtonProgressViewStyle.swift

```diff
@@ -0,0 +1,28 @@
+import SwiftUI
+import WooFoundation
+
+/// A view that displays an animated progress indicator with a circular shape.
+struct POSButtonProgressViewStyle: ProgressViewStyle {
+    private let size: CGFloat
+    private let lineWidth: CGFloat
+
+    /// - Parameters:
+    ///   - size: dimension of the whole progress view.
+    ///   - lineWidth: thickness of the circular progress indicator.
+    init(size: CGFloat, lineWidth: CGFloat) {
+        self.size = size
+        self.lineWidth = lineWidth
+    }
+
+    func makeBody(configuration: Configuration) -> some View {
+        ProgressView(configuration)
+            .progressViewStyle(IndefiniteCircularProgressViewStyle(
```

Comment:
It's a nice touch that we use the same custom view for both the into to POS and loading buttons 💯